lib/ansible/plugins/lookup/csvfile.py

```diff
@@ -192,23 +142,19 @@ def run(self, terms, variables=None, **kwargs):
             key = kv['_raw_params']
 
             # parameters override per term using k/v
-            try:
-                reset_params = False
-                for name, value in kv.items():
-                    if name == '_raw_params':
-                        continue
-                    if name not in paramvals:
-                        raise AnsibleAssertionError('%s is not a valid option' % name)
-
-                    self._deprecate_inline_kv()
-                    self.set_option(name, value)
-                    reset_params = True
+            reset_params = False
+            for name, value in kv.items():
+                if name == '_raw_params':
+                    continue
+                if name not in paramvals:
+                    raise AnsibleAssertionError(f'{name!r} is not a valid option')
```

Comment:
It looks like this can just be `ValueError`:

```suggestion
                    raise ValueError(f'{name!r} is not a valid option.')
```

That also means the `AnsibleAssertionError` import can be removed.